### PR TITLE
Periodic Boundary Layer implementation

### PR DESCRIPTION
**Pull request by slaizet**

### CODE REVIEW

File: src/variables.f90 [hunk 1]
```diff
@@ -1276,6 +1291,18 @@ subroutine init_variables
     else
        call stretching()
 
+       ! compute integral weights for stretched mesh - Ricardo Frantz
```
Comment (mathrack) [inline src/variables.f90:1294]: The channel case contains the exact integration procedure for streched meshes. This is an approximation. Please provide the motivation for using the approximate integration and not the exact one.

File: src/parameters.f90 [hunk 2]
```diff
@@ -566,6 +640,15 @@ subroutine parameter(input_i3d)
      if (itype==itype_gravitycur) then
         write(*,*)  "Initial front location: ", pfront
      endif
+     ! Check output parameters are valid for TTBL
+     if (itype.eq.itype_ptbl) then
+        if (ioutput < ilist .or. mod(ioutput, ilist) /= 0) then
+           if (nrank == 0) write (*, *) 'ioutput must be exactly divisible by ilist'
```
Comment (mathrack) [inline src/parameters.f90]: Calling `decomp_2d_abort` is probably better

File: src/variables.f90 (hunk 1 above)
Comment (slaizet) [reply src/variables.f90:1294]: no motivation (I did not do it) but will look into implementing the exact one

File: src/parameters.f90 (hunk 2 above)
Comment (slaizet) [reply src/parameters.f90]: we use mpi_abort in various places, what would be the benefit of using decomp_2d_abort here?

Comment (mathrack) [reply src/parameters.f90]: Calling the subroutine `decomp_2d_abort` takes one line and it prints more stuff :
```
call decomp_2d_abort(__FILE__, __LINE__, ioutput, "ioutput must be exactly divisible by ilist")
```
the master rank will print the name of the file containing the error, the line where the error is located, the value of `ioutput` and the error message to both stdout and stderr.

It is shorter to use decomp_2d_abort and it provides more information
https://github.com/2decomp-fft/2decomp-fft/blob/dev/src/decomp_2d_mpi.f90#L154

Comment (slaizet) [reply src/parameters.f90]: fixed

**Pull request merged**